### PR TITLE
feat: adds script.realmCreated

### PR DESCRIPTION
**Pull request by Lightning00Blade**

### CODE REVIEW

File: src/bidiMapper/bidiMapper.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export type {
   ICdpConnection as CdpConnection,
 } from '../cdp/cdpConnection.js';
 export type {ICdpClient, ICdpClient as CdpClient} from '../cdp/cdpClient.js';
+export {EventEmitter} from '../utils/EventEmitter.js';
```
Comment (Lightning00Blade) [inline src/bidiMapper/bidiMapper.ts:29]: This is needed as Puppeteer EventEmitter is incomparable with BiDi Mapper EventEmitter.

File: tests/test_realm.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    # Wait for `script.realmCreated` event.
+    assert {
+        "method": "script.realmCreated",
+        "params": {
+            "type": "window",
+            "origin": "null",
+            "realm": ANY_STR,
+            "context": context_id,
+        }
+    } == response[0]
```
Comment (Lightning00Blade) [inline tests/test_realm.py:47]: Help needed here. The response is a tuple with 1 element.

File: src/bidiMapper/bidiMapper.ts (hunk 1 above)
Comment (thiagowfx) [reply src/bidiMapper/bidiMapper.ts:29]: Can this be `export type`, though?

File: src/protocol/protocol.ts [hunk 3]
```diff
@@ -787,7 +787,18 @@ export namespace Script {
...
+  >;
+
+  export type RealmCreated = {
+    method: 'script.realmCreated';
+    params: RealmInfo;
+  };
+
   export enum EventNames {
+    RealmCreated = 'script.realmCreated',
     MessageEvent = 'script.message',
```
Comment (thiagowfx) [inline src/protocol/protocol.ts:802]: Please sort the event names

Could also wrap `// keep-sorted` around here, but not absolutely needed as there are only two entries

File: tests/test_realm.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Google LLC.
```
Comment (thiagowfx) [inline tests/test_realm.py:1]: Should this be in `test_browsing_context.py` instead? @sadym-chromium thoughts?

File: src/bidiMapper/bidiMapper.ts (hunk 1 above)
Comment (Lightning00Blade) [reply src/bidiMapper/bidiMapper.ts:29]: Types are different, I am happy if we align with something external - Node EventEmitter and we use that as a CommonInterface. 
Else we can't as you extend the interface CDPClient with the class and that brakes Typescript as its looking for `#private` which can't be provided.

File: tests/test_realm.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    # Wait for `script.realmCreated` event.
+    assert {
+        "method": "script.realmCreated",
+        "params": {
+            "type": "window",
+            "origin": "null",
+            "realm": ANY_STR,
+            "context": context_id,
+        }
+    } == response[0]
```
Comment (thiagowfx) [inline tests/test_realm.py:48]:
```suggestion
    # Wait for `script.realmCreated` event.
    assert ({
        "method": "script.realmCreated",
        "params": {
            "type": "window",
            "origin": "null",
            "realm": ANY_STR,
            "context": context_id,
        }
    },) == response
```

File: tests/test_realm.py (hunk 2 above)
Comment (thiagowfx) [reply tests/test_realm.py:48]: See suggestion below

File: src/bidiMapper/bidiMapper.ts (hunk 1 above)
Comment (thiagowfx) [reply src/bidiMapper/bidiMapper.ts:29]: Sure, we could look into that in another PR.

**Review by sadym-chromium — APPROVED** [review]

**Pull request merged**